domain/access/state/permission.go

```diff
@@ -294,22 +304,39 @@ func (st *PermissionState) ReadAllUserAccessForUser(ctx context.Context, subject
 	}
 	var (
 		permissions []dbReadUserPermission
-		user        dbPermissionUser
+		users       []dbPermissionUser
 	)
-	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		user, err = st.findUserByName(ctx, tx, subject)
-		if err != nil {
-			return errors.Trace(err)
-		}
+	query := `
+WITH what AS (
+    select 'cloud' AS type, name AS grant_on FROM cloud
+    UNION
+    select 'model' AS type, uuid AS grant_on FROM model_list
+    UNION
+    select 'controller' AS type, 'controller' AS grant_on
+)
+SELECT (u.uuid, u.name, u.display_name, u.created_at, u.disabled) AS (&dbPermissionUser.*),
+       creator.name AS &dbPermissionUser.created_by_name,
+       w.type AS &dbReadUserPermission.object_type,
+       (p.uuid, p.grant_on, p.grant_to, p.access_type) AS (&dbReadUserPermission.*)
+FROM   v_user_auth u
+       JOIN user AS creator ON u.created_by_uuid = creator.uuid
+       JOIN v_permission p ON u.uuid = p.grant_to
+       JOIN what AS w ON p.grant_on = w.grant_on
+WHERE  u.removed = false
+       AND u.name = $M.name
+`
 
-		userPermissions, err := st.readUsersPermissions(ctx, tx, user.UUID)
-		if err != nil {
-			return errors.Trace(err)
-		}
+	queryStmt, err := st.Prepare(query, dbPermissionUser{}, dbReadUserPermission{}, sqlair.M{})
```

Comment:
Something Gustavo suggested the other day was that we should be embracing types more and relying less on maps for ad-hoc parameterisation.

You've got 3 (at least) uses of single entry map with "name" in this file. We might as we create a type for it and deliver name args with it.

Reply:
done